hayeon/programmers-1/예산.js

```diff
@@ -0,0 +1,33 @@
+function solution(d, budget) {
+  let result = 0;
+  let sum = 0;
+
+  d = d.sort((a, b) => a - b);
+
+  for (let i = 0; i < d.length; i++) {
+    sum += d[i];
+
+    if (sum <= budget) {
+      result++;
+    } else {
+      break;
+    }
+  }
+  return result;
+}
```

Comment:
```suggestion
function solution(d, budget) {
  let result = 0;
  let sum = 0;

  d.sort((a, b) => a - b);

  for (let i = 0; i < d.length; i++) {
    sum += d[i];

    if (sum > budget) return result;
    result++;
  }
    
  return result;
}
```
- 고생하셨습니다!! 🙆🏻‍♂️ break문을 통해 for문 순회를 종료하는 아이디어도 좋았습니다!! 하지만 단순히 sum이 budget보다 크면 곧바로 return result를 해주는 것도 좋은 방법이 될 것 같습니다.
- 그리고 sort메서드는 기존 배열을 바꿔주는 메서드입니다. 따라서 다시 d에다 할당 할 필요가 없습니다.
```js
const arr = [5, 2, 3, 1, 4];

arr.sort((a,b) => a - b);

console.log(arr); // [1, 2, 3, 4, 5]
```